### PR TITLE
Enhanced dba functions, several replaced database calls

### PR DESCRIPTION
**Pull request by annando**
The database functions "insert" and "update" are modified to better handle the problem of unique key problems.

Additionally the old "q" function was replaced at several locations with the new "dba" functions. The new functions can simplify the existing code.

### CODE REVIEW

File: include/group.php [hunk 1]
```diff
@@ -144,12 +140,7 @@ function group_add_member($uid,$name,$member,$gid = 0) {
 				// we indicate success because the group member was in fact created
 				// -- It was just created at another time
  	if (! dbm::is_result($r)) {
-		$r = q("INSERT INTO `group_member` (`uid`, `gid`, `contact-id`)
-			VALUES( %d, %d, %d ) ",
-			intval($uid),
-			intval($gid),
-			intval($member)
```
Comment (tobiasd) [inline include/group.php]: das ist jetzt hier nur exemplarisch, aber ich gehe davon aus, dass das weggelassene intval Magisch von dba ausgeführt wird oder?

Comment (annando) [inline include/group.php]: Jepp!

And for the English speaking community: There is no need to do escaping with prepared statements - which are used for the new database functions.

Comment (tobiasd) [inline include/group.php]: :-) cLI

**Comment by annando** [comment]
I had to done a final commit that improves the parameter handling after it was approved.

**Pull request merged**